docs/plugin-authoring-evaluator.md

```diff
@@ -0,0 +1,380 @@
+
+Here are a few examples:
+
+Output:
+Chicken parm sandwich
+Response:
+{ "reason": "This is a classic sandwich enjoyed by many - totally delicious", "verdict":"yes"}
+
+Output:
+Boston logan international airport tarmac
+Response:
+{ "reason": "This is not edible and definitely not delicious.", "verdict":"no"}
+
+Output:
+A juicy peice of gossip
+Response:
+{ "reason": "Gossip is sometimes metaphorically referred to as tasty.", "verdict":"maybe"}
+
+Here is a new submission to assess:
+
+Output:
+{{output}}
+Response:
+`;
+```
+
+#### Define the scoring function
+
+Now, define the function that will take an example which includes `output` as is required by the prompt and score the result. Genkit test cases include `input` and `output` as required fields, with optional fields for `context`. It is the responsibility of the evaluator to validate that all fields required for evaluation are present.
+
+This example leverages `handlebars` to hydrate the prompt and `zod` to format and validate the response.
+
+```ts
+// Define the expected output values
+const DELICIOUSNESS_VALUES = ['yes', 'no', 'maybe'] as const;
+
+// Define the response schema expected from the LLM
+const DeliciousnessDetectionResponseSchema = z.object({
+  reason: z.string(),
+  verdict: z.enum(DELICIOUSNESS_VALUES),
+});
+
+type DeliciousnessDetectionResponse = z.infer<
+  typeof DeliciousnessDetectionResponseSchema
+>;
+
+/**
+ * Score an individual test case for delciousness.
+ */
+export async function deliciousnessScore<
+  CustomModelOptions extends z.ZodTypeAny,
+>(
+  judgeLlm: ModelArgument<CustomModelOptions>,
+  dataPoint: BaseDataPoint,
+  judgeConfig?: CustomModelOptions
+): Promise<Score> {
+  const d = dataPoint;
+  try {
+    // Validate the input has required fields
+    if (!d.output) {
+      throw new Error('Output is required for Funniness detection');
+    }
+
+    //Hydrate the prompt
+    const promptTemplate = Handlebars.compile(QUESTION_GEN_PROMPT);
+    const finalPrompt = promptTemplate({
+      output: d.output,
+    });
+
+    // Call the LLM to generate an evaluation result
+    const response = await generate({
+      model: judgeLlm,
+      prompt: finalPrompt,
+      config: judgeConfig,
+    });
+
+    // Parse the output
+    const parsedResponse = extractJson<DeliciousnessDetectionResponse>(
+      response.text()
+    );
+    if (!parsedResponse) {
+      throw new Error(`Unable to parse evaluator response: ${response.text()}`);
+    }
+
+    // Return a scored response
+    return {
+      score: parsedResponse.verdict,
+      details: { reasoning: parsedResponse.reason },
+    };
+  } catch (err) {
+    console.debug(
+      `Deliciousness evaluation failed with error ${err} for sample ${JSON.stringify(
+        d
+      )}`
+    );
+    throw err;
+  }
```

Comment:
Consider removing this log and re-throw. If an error happens, the developer can find the details of the error by looking at the trace in the Dev UI.

Reply:
done.